Source/Applications/SystemCenter/Model/Security/SecurityGroup.cs

```diff
@@ -278,7 +283,31 @@ protected override DataTable GetSearchResults(PostData postData, int? page)
                 row["DisplayName"] = group.DisplayName;
             }
 
-            // #ToDo Add Filtering by DisplayName and Type
+            DataTable filteredDataTable = dataTable.Clone();
+            IEnumerable<DataRow> filteredRows = dataTable.AsEnumerable();
+            foreach (Search search in postData.Searches)
```

Comment:
It has to clone all the columns in order to keep the structure of the table so that the filtered table can later be set to the original data table. It doesn't clone the data, only the structure of the table.